pkg/controller/common/helpers.go

```diff
@@ -330,6 +329,16 @@ func ValidateMachineConfig(cfg mcfgv1.MachineConfigSpec) error {
 		if err := ValidateIgnition(ignCfg); err != nil {
 			return err
 		}
+
+		typedConfig, err := ConvertConfig(ignCfg)
```

Comment:
Compression field is part of ignition spec.
As per https://github.com/openshift/machine-config-operator/pull/2820#issuecomment-963387477 we are leaning towards moving supported ignition config changes in MCO to controller, shouldn't this logic go into ValidateIgnition()?

Reply:
I originally put checking compression in `ValidateIgnition` but @yuqi-zhang said to move it to here, because `ValidateIgnition` is more about checking whether the Ignition is in the correct format

Reply:
Right, that and the fact that we call validateIgnition in the daemon, whereas validateMachineConfig only happens in the controller, so I thought it would make more sense to have the controller be the gate here.

If we feel like we should have a clearer separation between ignition and MC only fields, I am also not against moving to validateIgnition

Reply:
My main reason for the change was to avoid confusion. ignition Config is at the top level part of MC, so it is more logical to put sub-field validation at one place.
What we can do is either:
1. Keep the existing code and add a comment with our future inetntion (to move all supported ignition config check from daemon to controller)
2. Move this new check into a new function called something like supportedIgnitionCheck().

Reply:
Don't have a strong preference, but I added a comment. Look okay?

Reply:
Like others said ValidateIgnition isn't checking the content, just the form. 

This type of check really belongs in reconcilable: https://github.com/openshift/machine-config-operator/blob/76d6155c1dab8050db4d911c32d3e7113faedbfd/pkg/daemon/update.go#L733

In reconcilable, we are checking whether a user only modified the supported ignition sections. This type of check belongs there. Moving functions from daemon to controller is out of scope for this BZ, , I would prefer to keep the code consistent and have this check in that function along with consistent messaging. We can move at a later date the whole thing, but it's important to be consistent until then.

Reply:
Okay @yuqi-zhang thought moving gradually in the right direction was more important than consistency

Reply:
If we have a function checking these types of things already, it doesn't make a lot of sense to move 1 check elsewhere like Sinny pointed out above.

There's not a great reason to do it piecemeal imo it's better to add the check where it currently belongs then plan for/move those checks correctly.

Reply:
Okay looked at `reconcilable`, but I think pretty much everything there contains stuff about both the old and new config. Is there anything that does need to move? Because I think that would mean the check I'm adding doesn't really belong there anyways - `reconcilable` is about whether changes can be done in place, whereas the check I wrote is a check for something we never support

Reply:
The checks there are a proxy for ignition fields/changes we don't support: https://github.com/openshift/machine-config-operator/blob/76d6155c1dab8050db4d911c32d3e7113faedbfd/pkg/daemon/update.go#L782

https://github.com/openshift/machine-config-operator/blob/master/docs/MachineConfigDaemon.md#supported-vs-unsupported-ignition-config-changes

Reply:
We do support them for the initial machine config though, right? Whereas this change never allows compression

Reply:
`If the MC is applied on day 1, Ignition correctly writes the uncompressed version of the file.`

Based on the [bug](https://bugzilla.redhat.com/show_bug.cgi?id=1970218#c0), this sounds like a day2 issue?

@bgilbert can clarify?

Reply:
It is a day 2 issue, but @yuqi-zhang said to disable it for day 1 as well. Do you think we should allow it day 1?

Reply:
So a few thoughts here:

The `reconcileable` check means whether we can go from one condition to the next. For example, you cannot go from condition A to condition B if disks were changed, since we cannot **make that change**, but we can allow that **if it was already on disk**.

Compression is a bit different in that. We are simply **not allowing it at all**. Even as day 1, since our MCO code doing a writefile will immediately overwrite it. This will cause issues in the future, since there is no way for us to manage a compressed file. We should reject it during installs as well, unless we want to create a file we can otherwise no longer manipulate.

I still strongly think we should be doing this in the controller, not the daemon.

Reply:
For example, we are validating `KernelType` already in the controller, and not in reconcileable: https://github.com/openshift/machine-config-operator/blob/5b4670f644e10a425ec9858cab25d56b5542a84c/pkg/controller/common/helpers.go#L324

Reply:
If this is the case, can this PR add a description explaining these details and implementation choice and also clarify on the actual BZ what work is going in and why so that it can be properly verified? Because otherwise it's really not clear.